examples/QubitRotation.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Qubit Rotation with JAX
+# 
+# In this notebook, we shall investigate how we can use simple gradient descent to optimize for the a qubit rotation. The idea is as follows: We start in $|1 \rangle$ state on a Bloch Sphere and apply rotations around Z-Y-Z axes (in that order), each of the three with angles $\phi$, $\theta$, and $\omega$ respectively
+# 
+# $R(\phi,\theta,\omega) = RZ(\omega)RY(\theta)RZ(\phi)= \begin{bmatrix}
+# e^{-i(\phi+\omega)/2}\cos(\theta/2) & -e^{i(\phi-\omega)/2}\sin(\theta/2) \\
+# e^{-i(\phi-\omega)/2}\sin(\theta/2) & e^{i(\phi+\omega)/2}\cos(\theta/2)
+# \end{bmatrix}$
+# 
+# to reach the target state, which is defined to be $|0 \rangle$ in this case.
+
+
+import jax.numpy as jnp
+import numpy as onp
+from qutip import *
+from jax import grad, jit
+from jax.random import normal, PRNGKey
+
+
+def rot(phi, theta, omega):
+    """Returns a rotation matrix describing rotation around Z-Y-Z axis
+    
+    Parameters:
+    ----------
+        phi: int/float 
+             rotation angle for the second rotation around z axis 
+             
+        theta: int/float
+             rotation angle around y axis
+             
+        omega: int/float
+             rotation angle for the first rotation around z axis
+        
+    Returns:
+    --------
+        rm: array[complex]
+             array representing the rotation matrix 
+    """
+    
+    cos = jnp.cos(theta/2)
+    sin = jnp.sin(theta/2)
+
+    return jnp.array(
+    [
+        [jnp.exp(-0.5j*(phi+omega))*cos, -(jnp.exp(0.5j*(phi-omega)))*sin],
+        [jnp.exp(-0.5j*(phi-omega))*sin, jnp.exp(0.5j*(phi+omega))*cos],
+        
+    ]
+    )
+
+
+def cost(phi, theta, omega, ket):
+    r"""Returns the fidelity between the evolved state and :math: `|0 \rangle` state
+    
+    Parameters:
+    ----------
+        phi: int/float 
+             rotation angle for the second rotation around z axis 
+             
+        theta: int/float
+             rotation angle around y axis
+             
+        omega: int/float
+             rotation angle for the first rotation around z axis
+             
+        ket: array[complex] 
+             array representing the ket to be acted upon by by the rotation matrix
+
+    Returns:
+    -------
+        float: 
+        fidelity between the :math:`|0 \rangle` state and the evolved state under rotation    
+    """
+    
+    evolved = jnp.dot(rot(phi, theta, omega), ket)
+    return jnp.absolute(jnp.vdot(evolved.T,basis(2,0).full()))
```

Comment:
You must use spaces after "," see pep8 recommendations.

Reply:
@araza6 please mark the conversations as resolved (if you see the option below, so that reviewers are sure this is taken care of) or comment each of these comments below.